File: chp3/python/Mocha.py
from CondimentDecorator import CondimentDecorator


class Mocha(CondimentDecorator):
    def __init__(self, beverage):
        super(Mocha, self).__init__()
        self.beverage = beverage

    def getDescription(self):
        return self.beverage.getDescription() + ", Mocha"

    def cost(self):
        return self.beverage.cost() + 0.20

File: chp3/java/src/Mocha.java
public class Mocha extends CondimentDecorator {
    public Beverage beverage;

    public Mocha(Beverage beverage) {
	this.beverage = beverage;
    }

    public String getDescription() {
	return beverage.getDescription() + ", Mocha";
    }

    public double cost() {
	return beverage.cost() + .20;
    }
}
